### ISSUE TITLE
Space before Definitions

### ISSUE THREAD

**Issue by mortymacs** [VidocqH/lsp-lens.nvim#20]
Hi,

In Go, it shows space before interfaces definitions:
![image](https://github.com/VidocqH/lsp-lens.nvim/assets/1261448/984b2bc9-c6ed-4626-9a47-b6b29f758dff)

As the screenshot shows, functions and methods display correctly, but only the interface has spaces.

Cheers,
Mort

**Comment by mortymacs** [comment]
I also saw it when I defined a type:
![image](https://github.com/VidocqH/lsp-lens.nvim/assets/1261448/7e2ef450-5b11-4f88-8b13-4b490d11dbdb)

**Comment by mcauley-penney** [comment]
I can reproduce the problem. Looking at the content of the document symbols (which are returned by the LSP itself), the start character for the range, which lsp-lens uses to figure out where to place the extmark lines, is actually 5. Notice in the image below that the range start value for the `Hello` function is 0. 

So, this appears to be a problem with Gopls, or lsp-lens has to figure out its own way to determine the indent for each extmark line it places.

![image](https://github.com/VidocqH/lsp-lens.nvim/assets/59481467/6a03b0fd-1caf-43ae-825b-8232941f2ffa)

**Comment by mortymacs** [comment]
Hey @mcauley-penney, thanks for looking into this issue and providing the details. I'm on board with supporting this feature in the lsp-lens itself since it might be useful for other languages too.

**Comment by VidocqH** [comment]
> Hey @mcauley-penney, thanks for looking into this issue and providing the details. I'm on board with supporting this feature in the lsp-lens itself since it might be useful for other languages too.

Sorry for replay late. I think it can be solved with `vim.fn.indent(lnum)`, always placed lsp-info at the start point of the given line, but I am not sure if this will broke with some languages.

**Comment by mortymacs** [comment]
Is it testable by something like this? `:lua vim.fn.indent(5)`

**Comment by mortymacs** [comment]
> > Hey @mcauley-penney, thanks for looking into this issue and providing the details. I'm on board with supporting this feature in the lsp-lens itself since it might be useful for other languages too.
> 
> Sorry for replay late. I think it can be solved with `vim.fn.indent(lnum)`, always placed lsp-info at the start point of the given line, but I am not sure if this will broke with some languages.

Is it testable by something like this? `:lua vim.fn.indent(5)`

**Comment by VidocqH** [comment]
> > > Hey @mcauley-penney, thanks for looking into this issue and providing the details. I'm on board with supporting this feature in the lsp-lens itself since it might be useful for other languages too.
> > 
> > 
> > Sorry for replay late. I think it can be solved with `vim.fn.indent(lnum)`, always placed lsp-info at the start point of the given line, but I am not sure if this will broke with some languages.
> 
> Is it testable by something like this? `:lua vim.fn.indent(5)`

Yes, this func returns how many indents the given line have. Namely the starting point of the given line.

**Comment by mortymacs** [comment]
I tried it, but still, I'm not sure what should I do in my LSP config. Would you please elaborate a little?

**Comment by VidocqH** [comment]
> I tried it, but still, I'm not sure what should I do in my LSP config. Would you please elaborate a little?

Sorry for the ambiguity, this func is to get the indent of the given line which is the starting point of the line, I mean maybe we can use this func to get the indent and always display lens info at that point rather than the point provided by LSP, this will eliminate all leading spaces but I am not sure if this will broke with some language. 🤔

**Comment by mortymacs** [comment]
> > I tried it, but still, I'm not sure what should I do in my LSP config. Would you please elaborate a little?
> 
> Sorry for the ambiguity, this func is to get the indent of the given line which is the starting point of the line, I mean maybe we can use this func to get the indent and always display lens info at that point rather than the point provided by LSP, this will eliminate all leading spaces but I am not sure if this will broke with some language. 🤔
> 
> Something like this: <img alt="截屏2023-10-27 上午1 54 41" width="624" src="https://user-images.githubusercontent.com/16725418/278433462-8d484724-799f-4142-acbe-45b1395d2d13.png">

No problem at all! ;)
That's a good idea! It's also good to make it configurable (e.g. `by_lsp` or `static`) something like:
```lua
....
require'lsp-lens'.setup({
  ...
  advanced = {
    indent = "by_lsp",
  },
})
```

Also, one unrelated question, how did you enable the third section (`icon Vidocq + 1`) in your info? That's very interesting :)

**Comment by VidocqH** [comment]
> > > I tried it, but still, I'm not sure what should I do in my LSP config. Would you please elaborate a little?
> > 
> > 
> > Sorry for the ambiguity, this func is to get the indent of the given line which is the starting point of the line, I mean maybe we can use this func to get the indent and always display lens info at that point rather than the point provided by LSP, this will eliminate all leading spaces but I am not sure if this will broke with some language. 🤔
> > Something like this: <img alt="截屏2023-10-27 上午1 54 41" width="624" src="https://user-images.githubusercontent.com/16725418/278433462-8d484724-799f-4142-acbe-45b1395d2d13.png">
> 
> No problem at all! ;) That's a good idea! It's also good to make it configurable (e.g. `by_lsp` or `static`) something like:
> 
> ```lua
> ....
> require'lsp-lens'.setup({
>   ...
>   advanced = {
>     indent = "by_lsp",
>   },
> })
> ```
> 
> Also, one unrelated question, how did you enable the third section (`icon Vidocq + 1`) in your info? That's very interesting :)

Okay, let it configurable is good LOL.

that section is `git_author`, I finished it two days ago and testing it for myself to see if there's a bug. Plan to publish it in next few days. 😂

**Comment by mortymacs** [comment]
I just saw it in my Lua codes too:
![image](https://github.com/VidocqH/lsp-lens.nvim/assets/1261448/08c3f34b-0345-4313-9cb9-6500478bb6cc)

**Comment by mortymacs** [comment]
I just found another situation when the indentation needs to be adjusted
![image](https://github.com/VidocqH/lsp-lens.nvim/assets/1261448/c33dd627-7522-40b4-8902-c9189e5addb5)
